libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1979,6 +1985,43 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         )
     }
 
+    /**
+     * Asynchronously passes in the current sensor data of the user.
+     *
+     * @param data The current sensor data of user
+     * @param callback Callback which is called when the async operation is completed
+     */
+    @ExperimentalPreviewMapboxNavigationAPI
+    fun updateExternalSensorData(data: SensorData, callback: UpdateExternalSensorDataCallback) {
+        navigator.updateExternalSensorData(data.toNativeSensorData()) {
+            callback.onResult(it)
+        }
+    }
+
+    /**
+     * Sets a callback for ADASIS messages
+     *
+     * @param adasisConfig Adasis config
+     * @param callback Message callback
+     */
+    @ExperimentalPreviewMapboxNavigationAPI
+    fun setAdasisMessageCallback(adasisConfig: AdasisConfig, callback: AdasisV2MessageCallback) {
```

Comment:
It's not a one-shot callback, right? If that's true, we call things like this observer, not callback.
Also maybe to register/unregister, as for all other observers we have. We also allow to unregister one specific observer.

Reply:
We do not support multiple callbacks for adasis (and probably don't have to). ADASIS generation is enabled once user sets callback with corresponding config. What if end up with methods like `startAdasisEhp(config, callback)` and `stopAdasisEhp()`?

Reply:
We already provide something like that. Ok, let's leave it as is for now.